packages/react/src/components/DetailsList/DetailsColumn.base.tsx

```diff
@@ -129,11 +148,7 @@ export class DetailsColumnBase extends React.Component<IDetailsColumnProps> {
                       <IconComponent className={classNames.iconClassName} iconName={column.iconName} />
                     )}
 
-                    {column.isIconOnly ? (
-                      <span className={classNames.accessibleLabel}>{column.name}</span>
-                    ) : (
-                      column.name
-                    )}
+                    {onRenderHeader()}
```

Comment:
This needs to pass in the props that were used, minimally the `column`, or else it's hard to supply an override.

Reply:
onRenderHeader is defined in the Render function, where column is this.props.column, so passing in this.props should be functionally equivalent to how it currently is

Reply:
Updated